openmc/surface.py

```diff
@@ -436,36 +448,44 @@ class ZPlane(Plane):
 
     Parameters
     ----------
-    surface_id : int
+    surface_id : int, optional
         Unique identifier for the surface. If not specified, an identifier will
         automatically be assigned.
-    boundary_type : {'transmission, 'vacuum', 'reflective', 'periodic'}
+    boundary_type : {'transmission, 'vacuum', 'reflective', 'periodic'}, optional
         Boundary condition that defines the behavior for particles hitting the
         surface. Defaults to transmissive boundary condition where particles
         freely pass through the surface.
-    z0 : float
-        Location of the plane
-    name : str
+    z0 : float, optional
+        Location of the plane. Defaults to 0.
+    name : str, optional
         Name of the plane. If not specified, the name will be the empty string.
 
     Attributes
     ----------
     z0 : float
         Location of the plane
+    boundary_type : {'transmission, 'vacuum', 'reflective', 'periodic'}
+        Boundary condition that defines the behavior for particles hitting the
+        surface.
+    coeffs : dict
+        Dictionary of surface coefficients
+    id : int
+        Unique identifier for the surface
+    name : str
+        Name of the surface
+    type : str
+        Type of the surface, e.g. 'x-plane'
```

Comment:
I think there is a copy-paste issue in that each of the `Surface` subclasses has a `type` that is "e.g., 'x-plane'" - perhaps remove the "e.g." and simply specify the type of the corresponding `Surface`.